### ISSUE TITLE
RealNVP can't reconstruct samples

### ISSUE THREAD

**Issue by csalcedo001** [bayesiains/nflows#37]
Hi!

I was playing with RealNVP and tried to reconstruct samples after a forward and backward pass through the flow. After training the network on moons dataset, I ran the following program

```
x, label = datasets.make_moons(1000, noise=.05)
x = torch.from_numpy(x.astype(np.float32))

with torch.no_grad():
    z, _ = model._transform(x, context=torch.nn.Identity()(None))
    x_hat, _ = model._transform(z, context=torch.nn.Identity()(None))
```

The following plots show the distribution of the points after each pass

Original x:
![realnvp_x](https://user-images.githubusercontent.com/27221651/111531654-d402cd00-8732-11eb-89ac-7150a0ee0873.png)

Normalized z:
![realnvp_z](https://user-images.githubusercontent.com/27221651/111531797-014f7b00-8733-11eb-8821-3e86b13cd850.png)

Reconstruction x_hat:
![realnvp_x_hat](https://user-images.githubusercontent.com/27221651/111531842-0ad8e300-8733-11eb-88ef-181b147284f7.png)

The forward pass looks great, since it forms a normal distribution. However, the backward pass doesn't reconstruct the input x_hat similar to x as expected. Any feedback on what is happening would be of great help. Thanks!

**Comment by csalcedo001** [comment]
Double-checking my code I just realized my program was performing two forward passes instead of a forward and backward pass. After changing the last line to `x_hat, _ = model._transform.inverse(z, context=torch.nn.Identity()(None))` the reconstruction was perfect.

Sorry for the false alarm, I'm closing this issue.
